### ISSUE TITLE
[LOW] [Splits] [$500] Chat - Member stays hidden if the reply comes in the IOU report

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#32870]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.11.1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Navigate to https://staging.new.expensify.com/
2. Log in with two different accounts that didn't have any contact with each other yet
3. Account B: Create an IOU with account A
4. Account A: Open the chat with the IOU
5. Account A: Tap on the received IOU
6. Account A: Write any message
7. Account B: Navigate between conversations (to reveal the hidden account)

## Expected Result:
The member should be revealed

## Actual Result:
Member stays hidden if the reply comes in the IOU report

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/a2ded882-a0d3-4da8-84b5-1b8ebe7436c5

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01cc37bbe9540440f6</li>
        <li>Upwork Job ID: 1734327335985864704</li>
        <li>Last Price Increase: 2024-01-01</li>
    </ul>
</details>

**Comment by dksingla** [comment]
We are not getting enough details from the server about user A
`{
            "onyxMethod": "merge",
            "key": "personalDetailsList",
            "value": {
                "16127502": {
                    "accountID": 16127502,
                    "avatar": "https:\/\/d2k5nsl2zxldvw.cloudfront.net\/images\/avatars\/default-avatar_1.png",
                    **"firstName": "",
                    "lastName": "",**
                    "status": null
                },
                "16127512": {
                    "accountID": 16127512,
                    "avatar": "https:\/\/d2k5nsl2zxldvw.cloudfront.net\/images\/avatars\/default-avatar_13.png",
                    "firstName": "",
                    "lastName": "",
                    "status": null,
                    "displayName": "diwakij420@gearstag.com",
                    "login": "diwakij420@gearstag.com",
                    "pronouns": "",
                    "timezone": {
                        "automatic": true,
                        "selected": "Asia\/Calcutta"
                    },
                    "payPalMeAddress": "",
                    "phoneNumber": "",
                    "validated": true
                }
            }
        },`

**Comment by keisyrzk** [comment]
Hm please find my recording. I tried multiple times on staging server with fresh account being the `User B`. Have not seen the `hidden` naming at all. Did I miss something in reproduction steps?
I tried both ways when splitting the bill:
(1) click "split" on the side of the user on search
<img width="374" alt="Zrzut ekranu 2023-12-14 o 11 12 55" src="https://github.com/Expensify/App/assets/18525360/5634936c-be47-42bb-ac2b-332b0b12f21d">

(2)
click "Request (...)" in user details
<img width="366" alt="Zrzut ekranu 2023-12-14 o 11 13 46" src="https://github.com/Expensify/App/assets/18525360/a44b6042-9909-4b59-a00d-66d422c7ba2e">

Recording:
https://github.com/Expensify/App/assets/18525360/1992cce0-2502-49f8-b53d-63729efa03d5

**Comment by abdulrahuman5196** [comment]
Reviewing now

**Comment by abdulrahuman5196** [comment]
@strepanier03 Isn't this the expected behaviour? Or are we changing the hidden feature?

**Comment by strepanier03** [comment]
Apologies for the delay here, I was OoO. I'll confirm the expected behavior and update everyone here.

**Comment by strepanier03** [comment]
### Expected behavior

We created a function in Auth to determine whether userA knows userB. The criteria are:
- if userB has ever requested money or sent an expense report to userA
- if userB has ever sent a message in a DM or groupDM that userA is a part of
- if userA and userB share access to at least one non-public workspace room or domain room

It looks like we need to fix this because userA replying to userB should establish the relationship between the accounts that allows for the user to be revealed.

**Comment by strepanier03** [comment]
Confirming wave/VIP [here](https://expensify.slack.com/archives/C066HJM2CAZ/p1703028546121129).

**Comment by abdulrahuman5196** [comment]
We are waiting on this https://github.com/Expensify/App/issues/32870#issuecomment-1863612993. I think folks are off this week, so could expect delays in responses.

**Comment by strepanier03** [comment]
@abdulrahuman5196 - You're correct, the holidays and OoO are impacting the response here. I checked and nothing on my post yet. I'll check tomorrow but I think it's safe to assume no update will come until next week.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by abdulrahuman5196** [comment]
> @abdulrahuman5196 - You're correct, the holidays and OoO are impacting the response here. I checked and nothing on my post yet. I'll check tomorrow but I think it's safe to assume no update will come until next week.

Same here

**Comment by strepanier03** [comment]
I've bumped the Slack thread for an update and will take action here as soon as I can.

**Comment by strepanier03** [comment]
All right, we've added this to a VIP with a Low priority. We expect this will have an Internal element to it to resolve but if you disagree @abdulrahuman5196 please chime in. ([expected behavior](https://github.com/Expensify/App/issues/32870#issuecomment-1863611053))

**Comment by strepanier03** [comment]
Since this is Low priority, I'm switching to weekly until it's picked up so that I don't have to update it every day.

**Comment by strepanier03** [comment]
No update, low priority.

**Comment by strepanier03** [comment]
All good Melvin.

**Comment by strepanier03** [comment]
Hmmm, not sure why the overdue here is being so sticky.

**Comment by strepanier03** [comment]
All good Melvin, low priority for the time being.

**Comment by abdulrahuman5196** [comment]
@strepanier03 Are we going to work on this? Seems the daily label is added on this?

**Comment by strepanier03** [comment]
@abdulrahuman5196 - It's not a priority, so likely not, but I'm supposed to keep it daily so that's why the label is applied.

**Comment by strepanier03** [comment]
Pausing on this as we discuss this VIP internally.

**Comment by strepanier03** [comment]
All good Melvin.

**Comment by strepanier03** [comment]
Low priority and will be worked on when it's more of a priority.

**Comment by strepanier03** [comment]
All good Melvin.

**Comment by strepanier03** [comment]
Waiting to be picked up.

**Comment by strepanier03** [comment]
Assigned to a project and waiting to be picked dup.

**Comment by strepanier03** [comment]
Still up for grabs.

**Comment by strepanier03** [comment]
Chatting internally on the next step and urgency.

**Comment by strepanier03** [comment]
[Discussing](https://expensify.slack.com/archives/C01SKUP7QR0/p1710211827720109) whether or not to ask an expert agency to take this one on.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by strepanier03** [comment]
Thanks for the update.

**Comment by strepanier03** [comment]
No update at this time. Low priority.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by strepanier03** [comment]
This isn't reproducible after three weeks, so per [SO](https://stackoverflowteams.com/c/expensify/questions/18717/18718#18718), I'm going to close.
